tests/integration/db.go

```diff
@@ -189,20 +190,52 @@ func setupNode(s *state, opts ...node.Option) (*node.Node, string, error) {
 		opts = append(opts, node.WithStoreType(node.MemoryStore))
 
 	default:
-		return nil, "", fmt.Errorf("invalid database type: %v", s.dbt)
+		return nil, fmt.Errorf("invalid database type: %v", s.dbt)
 	}
 
 	if s.kms == PubSubKMSType {
 		opts = append(opts, node.WithKMS(kms.PubSubServiceType))
 	}
 
+	netOpts := make([]net.NodeOpt, 0)
+	for _, opt := range opts {
+		if opt, ok := opt.(net.NodeOpt); ok {
+			netOpts = append(netOpts, opt)
+		}
+	}
+
+	if s.isNetworkEnabled {
+		var addresses []string
+		for _, node := range s.nodes {
+			addresses = append(addresses, node.Peer.PeerInfo().String())
+		}
+		netOpts = append(netOpts, net.WithListenAddresses(addresses...))
```

Comment:
todo: `WithListenAddresses` is an option to define the node address. What you are doing here is assigning all the nodes known addresses to the new node. Using this option in test is only relevant when restarting a node and the previously known address of that specific node should be used.

That may require adding the address info to the `nodeState` for future usage.

Reply:
applied your proposal